src/js/App/analytics/segment-analytics.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { createContext, useContext, useEffect, useRef } from 'react';
+import { AnalyticsBrowser } from '@segment/analytics-next';
+import { isProd } from '../../utils';
+import { useSelector } from 'react-redux';
+import { AnyObject } from '../../types';
+
+type SegmentEnvs = 'dev' | 'prod';
+type SegmentModules = 'openshift';
+
+const KEY_FALLBACK = {
+  prod: 'nm7VsnYsBVJ9MqjaVInft69pAkhCXq9Q',
+  dev: 'Aoak9IFNixtkZJRatfZG9cY1RHxbATW1',
+};
+
+function getAdobeVisitorId() {
+  const visitor = window?.s?.visitor;
+  if (visitor) {
+    return visitor.getMarketingCloudVisitorID();
+  }
+
+  return -1;
+}
+
+const getAPIKey = (env: SegmentEnvs = 'dev', module: SegmentModules) =>
+  ({
+    prod: {
+      openshift: 'z3Ic4EtzJtHrhXfpKgViJmf2QurSxXb9',
+    },
+    dev: {
+      openshift: 'A8iCO9n9Ax9ObvHBgz4hMC9htKB0AdKj',
+    },
+  }[env]?.[module] || KEY_FALLBACK[env]);
+
+const registerUrlObserver = () => {
+  /**
+   * We ignore hash changes
+   * Hashes only have frontend effect
+   */
+  let oldHref = document.location.href.replace(/#.*$/, '');
+
+  window.onload = function () {
+    const bodyList = document.body;
+    const observer = new MutationObserver(function (mutations) {
+      mutations.forEach(function () {
+        const newLocation = document.location.href.replace(/#.*$/, '');
+        if (oldHref !== newLocation) {
+          oldHref = newLocation;
+          window?.sendCustomEvent('pageBottom');
+          setTimeout(() => {
+            window.segment?.page();
+          });
+        }
+      });
+    });
+    const config = {
+      childList: true,
+      subtree: true,
+    };
+    observer.observe(bodyList, config);
+  };
+};
+
+export const SegmentContext = createContext<{ ready: boolean; analytics?: AnalyticsBrowser }>({
+  ready: false,
+  analytics: undefined,
+});
+
+export type SegmentProviderProps = {
+  activeModule: string;
+};
+
+export const SegmentProvider: React.FC<SegmentProviderProps> = ({ activeModule, children }) => {
+  const analytics = useRef<AnalyticsBrowser>();
+  const user = useSelector(({ chrome: { user } }: AnyObject) => user);
+  useEffect(() => {
+    registerUrlObserver();
```

Comment:
shouldn't we `disconnect` it when the app is unmounted?

Reply:
If chrome is unmounted it means the tab was closed or the site has changed. There is no point in unregistering. It will be lost in a shadow realm :ghost:

Reply:
What about the concurrent mode? 😏

Reply:
Somehow Pendo in our previous setup seemed like it continued tracking AFTER someone left our domain. We had pageview data coming in from hundreds/thousands of other domains. A few of them could be explained by our Pendo config/keys getting copy/pasted and used on other websites, but there were so many that it seemed like the impossible was happening and our tracking was continuing to run in the browser after someone left our site. Just throwing that out there in case the shadow realm isn't shadowy enough (maybe some browsers have a bug or maybe Pendo's interface was incorrectly telling us what domain data had been captured from).

Reply:
Hmm, interesting. The only thing I can think of would be some weird data in local storage. This is a simple DOM observer triggered on a DOM change that cannot exist after the tree is destroyed. I will unregister it though just to be sure.